src/pages/front-page.tsx

```diff
@@ -3,45 +3,49 @@ import React from 'react';
 import { Text } from '../components/text';
 import { Spinner } from '../components/spinner';
 import { Table, RowType } from '../components/table';
+import { InfiniteScroll } from '../components/infinite-scroll';
 
 import { useUsers } from '../graphql/queries/use-users';
 
 import { translations } from '../helpers/translations';
 
 const frontPageTranslations = translations.pt.front_page;
+const loadNumber = 10;
 
 export const FrontPage = () => {
+  const [rows, setRows] = React.useState<RowType[]>([]);
+  const [page, setPage] = React.useState(0);
   const [isLoading, setIsLoading] = React.useState(false);
 
   const header = ['Nome', 'Email'];
-  const [rows, setRows] = React.useState<RowType[]>([]);
 
   const loadUsers = async () => {
     setIsLoading(true);
-    const { data } = await useUsers();
-    const rows: RowType[] = data.users.nodes.map((user: RowType) => ({
+    const { data } = await useUsers(page);
+    const newRows: RowType[] = data.users.nodes.map((user: RowType) => ({
       name: user.name,
       email: user.email,
     }));
-    setRows(rows);
+    setRows(newRows);
     setIsLoading(false);
   };
 
-  React.useEffect(() => {
+  const loadMoreUsers = () => {
```

Comment:
```suggestion
  const handleBottomHit = () => {
```

Pra vc já ir se acostumando com a convenção: todos esses callbacks de reação do usuário nós usamos a abordagem declarativa (o nome remete ao trigger dessa função), e não a abordagem imperativa, que é mais comum (o que essa função faz).